File: src/AndroidXamarinBenchmark/AndroidXamarinBenchmark/Activities/ActivityControlsVisualPerformance.cs
using Android.App;
using Android.OS;
using Android.Util;
using Android.Widget;
using System.Diagnostics;

namespace AndroidXamarinBenchmark.Activities
{

    [Activity(Label = "ActivityControlsVisualPerformance")]
    public class ActivityControlsVisualPerformance : Activity
    {

        private long _timeElapsed { get; set; }

        private long _startTime { get; set; }


        private Stopwatch Stopwatch;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetContentView(Resource.Layout.ActivityControlsVisualPerformance);

            LinearLayout linearLayoutWrapper = FindViewById<LinearLayout>(Resource.Id.linearLayout_wrapper);

            Stopwatch = new Stopwatch();
            Stopwatch.Start();

            //TODO: test Out of memory
            for (int j = 0; j < 450; j++)
            {
                LinearLayout linearLayout = new LinearLayout(this);
                LinearLayout.LayoutParams linearLayoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WrapContent, LinearLayout.LayoutParams.WrapContent);
                linearLayout.Orientation = Orientation.Horizontal;
                linearLayoutWrapper.AddView(linearLayout, linearLayoutParams);

                for (int i = 0; i < 20; i++)
                {
                    Button button = new Button(this);
                    button.Text = "X";
                    button.Click += Button_Click;
                    linearLayoutParams = new LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.WrapContent);
                    linearLayoutParams.Weight = 1;
                    linearLayout.AddView(button, linearLayoutParams);
                }
            }
        }

        public override void OnWindowFocusChanged(bool hasFocus)
        {
            base.OnWindowFocusChanged(hasFocus);
            if (_timeElapsed == 0)
            {
                Log.Info("benchmark", "ActivityControlsVisualPerformance Finished");
                Stopwatch.Stop();
                _timeElapsed = Stopwatch.ElapsedMilliseconds;
            }
        }

        private void Button_Click(object sender, System.EventArgs e)
        {
            new AlertDialog.Builder(this)
                    .SetTitle("Time elapsed")
                    .SetMessage(_timeElapsed + " miliseconds")
                    .SetNeutralButton("", (s, e2) => { })
                    .Show();
        }
    }

}

File: src/AndroidNativeBenchmark/app/src/main/java/com/rochapires/admin/androidnativebenchmark/Activities/ActivityControlsVisualPerformance.java
package com.rochapires.admin.androidnativebenchmark.Activities;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.ListView;

import com.rochapires.admin.androidnativebenchmark.R;


public class ActivityControlsVisualPerformance extends AppCompatActivity implements View.OnClickListener {

    private long timeElapsed;
    private long startTime;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_activity_controls_visual_performance);

        LinearLayout linearLayoutWrapper = (LinearLayout) findViewById(R.id.linearLayout_wrapper);

        startTime = System.currentTimeMillis();
        //TODO: test Out of memory
        for (int j = 0; j < 450; j++)
        {
            LinearLayout linearLayout = new LinearLayout(this);
            LinearLayout.LayoutParams linearLayoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
            linearLayout.setOrientation(LinearLayout.HORIZONTAL);
            linearLayoutWrapper.addView(linearLayout, linearLayoutParams);

            for (int i = 0; i < 20; i++)
            {
                Button button = new Button(this);
                button.setText("X");
                button.setOnClickListener(this);
                linearLayoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
                linearLayoutParams.weight = 1;
                linearLayout.addView(button, linearLayoutParams);
            }
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_activity_visual_performance, menu);
        return true;
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if(this.timeElapsed == 0){
            this.timeElapsed = System.currentTimeMillis() - startTime;
        }
    }

    @Override
    public void onClick(View view) {
        new AlertDialog.Builder(this)
                .setTitle("Time elapsed")
                .setMessage(timeElapsed + " miliseconds")
                .setNeutralButton("OK", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {

                    }
                })
                .show();
    }
}
